tests/behat/features/bootstrap/Page/SearchPage.php

```diff
@@ -0,0 +1,1402 @@
+    if (!$this->expectedResultsCount) {
+      $this->expectedResultsCount = new Stack();
+    }
+    return $this->expectedResultsCount->get($pop);
+  }
+
+
+  /**
+   * GetLargetsFacetAndClickit.
+   *
+   * @param array $facets
+   *    List of elements containing facets.
+   * @param bool $select
+   *    Default true - if true sets expected result. If false, pops the last value.
+   *
+   * @return string
+   *    Nonempty if failure.
+   *
+   * @throws Exception
+   *    In case of error.
+   */
+  private function getLargestFacetAndClickIt($facets = array(), $select = true) {
+    // Now we traverse them to find out how many results they 'promise' if selected.
+    $largestName = "";
+    $largestCount = 0;
+    $largestCheckbox = $facets[0];
+    foreach ($facets as $chkbox) {
+      // This can be a getElement.
+      $lcntElement = $chkbox->find('css', 'span.count');
+      if ($lcntElement) {
+        $lcnt = $lcntElement->getText();
+      }
+      else {
+        return "Did not find correct structure of facet. Should contain a 'span class=count' element.";
+      }
+      // Remove the paranthesises to retrieve the number and check if this is the largest number
+      // found this far. If so, we will save it for later use.
+      $lcnt = substr($lcnt, 1, strlen($lcnt) - 2);
+      if ($lcnt > $largestCount) {
+        $largestCount = $lcnt;
+        $largestName = $chkbox->find('css', 'a')->getText();
+        $largestCheckbox = $chkbox;
+      }
+    }
+    if (!$largestCheckbox) {
+      return "Did not find a largest facet to choose.";
+    }
+
+    // Save the 'promise' to enable later checks and tell the tester what we attempt, so
+    // if this fails there is a chance to interpretate the screenshot.
+    if ($select) {
+      $this->setExpectedSearchResultSize($largestCount);
+    }
+    else {
+      if ($this->expectedResultsCount->pop() < 0) {
+        return "Something went wrong with expected results - are you trying to deselect with selecting facets first?";
+      };
+    }
+    $this->logMsg(true, "Clicks facet: " . $largestName . " to get " . $this->expectedResultsCount->get() . " results.");
+
+    // Now we will select the largest one we ended up with.
+    $link = $largestCheckbox->find('css', 'a');
+    if ($link) {
+      $this->scrollTo($link);
+      $link->click();
+      $this->waitForPage();
+    }
+    else {
+      return "Expected to find a link in the html structure under a facet.";
+    }
+    return "";
+  }
+
+  /**
+   * Get Current Max Page Traversel setting.
+   *
+   * @return int
+   *    Set the maximum number of pages we want to traverse to scrape off the search result.
+   */
+  public function getMaxPageTraversals() {
+    return $this->maxPageTraversals;
+  }
+
+  /**
+   * Returns and resets the log messages. They can only be read once.
+   *
+   * @return string
+   *    The logmessages.
+   */
+  public function getMessages() {
```

Comment:
Not a fan of hidden side effects. Either split function in two, or add a parameter to indicate if log should be cleared

Reply:
Renamed to getAndClearMessages, as that is not a side effect, but the actual idea with this function, that it clears the stack of messages.